src/uart.rs

```diff
@@ -217,6 +218,32 @@ pub mod config {
     }
 
     impl SourceClock {
+        pub const fn default() -> Self {
+            #[cfg(not(esp_idf_version_major = "4"))]
+            const DEFAULT: uart_sclk_t = soc_periph_uart_clk_src_legacy_t_UART_SCLK_DEFAULT;
+            #[cfg(esp_idf_version_major = "4")]
+            const DEFAULT: uart_sclk_t = uart_sclk_t_UART_SCLK_APB;
+            Self::from_raw(DEFAULT)
+        }
+
+        pub const fn from_raw(source_clock: uart_sclk_t) -> Self {
+            match source_clock {
+                #[cfg(any(
+                    esp_idf_soc_uart_support_apb_clk,
```

Comment:
I just copied these from the `Default` impl, but I can certainly try replacing them.

Reply:
I honestly don't remember how they ended up there, but I'm pretty sure they simply don't work. We can also leave it there for a future PR - up to you.

Reply:
Nah I can finish it, but it will be tomorrow.

Reply:
I ran `cargo b --target xtensa-esp32-espidf -vv` to see the build script output, it does indeed expose those cfgs:

![image](https://github.com/esp-rs/esp-idf-hal/assets/23152057/18f0bf02-016e-4b7b-a60c-46ad4c770a25)

This is because it is part of the soc `Kconfig`:

![image](https://github.com/esp-rs/esp-idf-hal/assets/23152057/c942180b-1719-41e2-bc23-4f37dbc8b1e2)

(`components/soc/<chip>/include/soc/Kconfig.soc_caps.in`)

Because of that, `esp-idf-sys` will pick it up and expose as `cfg`s.

Reply:
Makes sense. I did not realize that these are part of the `Kconfig`!